### PR TITLE
Add dry-run mode to test.pl

### PR DESCRIPTION
**Pull request by smowton**
This gives [OK] status for tests that will be run and [SKIPPED] otherwise. Useful
for tools that want to inspect tests that ran / will run. User is out-of-tree, but hopefully this is uncontroversially a useful thing to be able to do anyhow.

### CODE REVIEW

File: regression/test.pl
Comment (tautschnig) [inline regression/test.pl]: A matter of taste: use `-n` instead of `-d`? I think that's more common for `-d` is mostly used for "debug".

File: regression/test.pl
Comment (tautschnig) [inline regression/test.pl]: Not sure this change is essential, but I'll leave it to you whether you feel this makes the message look neater.

**Review by tautschnig — APPROVED** [review]
See comments for matter-of-taste changes. I'll leave it to you to apply them or not, and once you're content just do the merge.

File: regression/test.pl
Comment (smowton) [reply regression/test.pl]: Oops, undone

File: regression/test.pl
Comment (smowton) [reply regression/test.pl]: Ahh, like `make -n`, good point. Done.

**Comment by reuk** [comment]
I don't know any Perl, so I don't think I'll be able to give a useful review. Happy to try if you can't find someone with more experience though

**Comment by smowton** [comment]
No worries, I'll tag @LAJW

**Review by LAJW — APPROVED** [review]
LGTM. Fairly straightforward change. Highlights the elegance of this wonderful language.

**Review by thk123 — APPROVED** [review]

**Pull request merged**
